src/OpenSage.Game/LanguageSetting.cs

```diff
@@ -0,0 +1,25 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using OpenSage.Data;
+using OpenSage.Utilities;
+
+namespace OpenSage
+{
+    public static class LanguageSetting
+    {
+        //do we need a default?
+        private const string DefaultLanguage = "englisch";
+        private static string _detectedLanguage;
+
+        public static string Current => string.IsNullOrEmpty(_detectedLanguage) ? DefaultLanguage : _detectedLanguage;
+
+        public static void ReadFromRegistry(IGameDefinition gameDefinition)
+        {
+            var languageRegistryKey = gameDefinition.RegistryKeys.FirstOrDefault(i => i.ValueName == "Language");
```

Comment:
As said in the other comment, `RegistryKeys` isn't really intended for this purpose.

What you should do instead is to either:

1. Just hardcode the registry key in this method (or something correspoding to it) and check if `IGameDefinition.Game` is `SageGame.CncGenerals` or `SageGame.CncGeneralsZeroHour`.
2. Add something like a `LanguageRegistryKey` to `IGameDefinition`. While that would technically be better (our goal is to have as little game-specific code in the engine as possible), it requires changing every single game definition and I'm not certain if there's an equivalent registry key for all (or even most) of the games.